starknet_py/utils/typed_data.py

```diff
@@ -137,19 +152,27 @@ def _encode_value(
 
         basic_type = BasicType(type_name)
 
-        if basic_type == BasicType.MERKLE_TREE and isinstance(value, list):
-            if context is None:
-                raise ValueError(f"Context is not provided for '{type_name}' type.")
-            return self._prepare_merkle_tree_root(value, context)
-
-        if basic_type in (BasicType.FELT, BasicType.SHORT_STRING) and isinstance(
-            value, (int, str, Revision)
-        ):
+        if (basic_type, self.domain.resolved_revision) in [
+            (BasicType.FELT, Revision.V0),
+            (BasicType.FELT, Revision.V1),
+            (BasicType.STRING, Revision.V0),
+            (BasicType.SHORT_STRING, Revision.V1),
+            (BasicType.CONTRACT_ADDRESS, Revision.V1),
+            (BasicType.CLASS_HASH, Revision.V1),
+        ] and isinstance(value, (int, str)):
             return int(get_hex(value), 16)
```

Comment:
Let's modify `get_hex` function to return `int`, also rename it to something like `parse_felt`.